core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -513,12 +518,54 @@ class HttpSamDAO(baseSamServiceURL: String, rawlsCredential: RawlsCredential, ti
       }
     }
 
+  override def listResourcesWithActions(resourceTypeName: SamResourceTypeName,
+                                        action: SamResourceAction,
+                                        ctx: RawlsRequestContext
+  ): Future[Seq[SamUserResource]] =
+    retry(when401or5xx) { () =>
+      val callback = new SamApiCallback[ListResourcesV2200Response]("listResourcesV2")
+
+      resourcesApi(ctx).listResourcesV2Async(
+        /* format = */ "hierarchical",
+        /* resourceTypes = */ util.List.of(resourceTypeName.value),
+        /* policies = */ util.List.of(),
+        /* roles = */ util.List.of,
+        /* actions = */ util.List.of(action.value),
+        /* includePublic = */ false,
+        callback
+      )
+
+      callback.future.map { resourcesResponse =>
+        resourcesResponse.getFilteredResourcesHierarchicalResponse
+          .getResources()
+          .asScala
+          .map { resource =>
+            SamUserResource(
+              resource.getResourceId,
+              toSamRolesAndActions(resource.getPolicies()), // TODO what to use here?
+              toSamRolesAndActions(resource.getPolicies()), // What are all these three things?
+              toSamRolesAndActions(resource.getPolicies()),
```

Comment:
Each policy in `getPolicies()` has info about whether it is public, direct, or inherited. I think you can use that info to fill in the details of `SamUserResource`, like:
```scala
          .map { resource =>
            val resourcePolicies = resource.getPolicies.asScala.toList
            val publicPolicies = resourcePolicies.filter(_.getIsPublic)
            // might need to filter public policies out of this next statement:
            val (inheritedPolicies, directPolicies) = resourcePolicies.partition(_.getInherited)

            SamUserResource(
              resource.getResourceId,
              toSamRolesAndActions(directPolicies.asJava), // TODO what to use here?
              toSamRolesAndActions(inheritedPolicies.asJava), // What are all these three things?
              toSamRolesAndActions(publicPolicies.asJava),
```
if you do this you probably also want to change `toSamRolesAndActions` to accept a Scala list instead of the repeated `.asJava` and `.asScala`